qiskit_ibm_runtime/qiskit_runtime_service.py

```diff
@@ -183,6 +206,42 @@ def __init__(
                 self._current_instance = self._get_hgp().name
                 logger.info("Default instance: %s", self._current_instance)
 
+    def _discover_backends_from_instance(self, instance: str) -> List[str]:
+        """Retrieve all backends from the given instance."""
+
+        # refactor this, is there an easier way to fetch all backends
+        # ntc 5779 would make things a lot faster - get list of backends
+        # from global search API call
+
+        # I don't think we should overwrite the api client like this
+        # but we can't access the backend config without the right instance
+        if self._account.instance != instance:
+            self._client_params.instance = instance
+            self._api_client = RuntimeClient(self._client_params)
+        return self._api_client.list_backends()
```

Comment:
You may have come up with a better solution to avoid overwriting the client, but in case it helps: I think that you could refactor the client_params object and/or the `RuntimeClient` object (I would have to look into the specifics) to be able to store more than 1 instance in the cloud case. This would allow to keep handles to all instances a user might need instead of imposing one instance at a time. They could be stored together with the backend name, so the client knows which instance to use to access a specific backend.